scopes/envs/envs/environments.main.runtime.ts

```diff
@@ -218,6 +218,19 @@ export class EnvsMain {
     throw new EnvNotFound(id, component.id.toString());
   }
 
+  /**
+   * get the env of the given component.
+   * This will try to use the regular getEnv but fallback to the calculate env (in case you are using it during on load)
+   * This is safe to be used on onLoad as well
+   */
+  getOrCalculateEnv(component: Component): EnvDefinition {
```

Comment:
can't we add this functionality to the existing getEnv?

Reply:
not exactly, there is a reason for this.
Anyway it's not part of this PR